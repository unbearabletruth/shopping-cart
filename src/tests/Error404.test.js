import { render, screen } from '@testing-library/react';
import Error404 from '../components/Error404';
import { MemoryRouter } from 'react-router-dom';

describe('Error404', () => {
    test('snapshot of error404 page', () => {
        const {container} = render(
            <MemoryRouter>
                <Error404 />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
      });

    test('renders 404 error with link', () => {
        render(
            <MemoryRouter>
                <Error404 />
            </MemoryRouter>
        );
        expect(screen.getByRole("link").textContent).toMatch(/Home/i);
        screen.getByText('404');
        screen.getByText('Sorry, but your page is not found');
    });
});