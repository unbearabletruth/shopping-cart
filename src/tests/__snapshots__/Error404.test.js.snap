// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 snapshot of error404 page 1`] = `
<div>
  <div
    id="errorWrapper"
  >
    <p
      id="errorNumber"
    >
      404
    </p>
    <p
      id="errorMessage"
    >
      Sorry, but your page is not found
    </p>
    <a
      href="/"
      id="errorToHome"
    >
      Home
    </a>
  </div>
</div>
`;
